business_requirement/models/business.py

```diff
@@ -0,0 +1,348 @@
+        required=True,
+        readonly=True,
+        default=lambda self: self.env.user,
+        states={
+            'draft': [('readonly', False)],
+            'confirmed': [('readonly', False)]}
+    )
+    confirmation_date = fields.Datetime(
+        string='Confirmation Date',
+        copy=False,
+        readonly=True
+    )
+    confirmed_user_id = fields.Many2one(
+        'res.users', string='Confirmed by',
+        copy=False,
+        readonly=True
+    )
+    responsible_user_id = fields.Many2one(
+        'res.users', string='Responsible',
+        copy=False,
+        readonly=True,
+        default=lambda self: self.env.user,
+        states={
+            'draft': [('readonly', False)],
+            'confirmed': [('readonly', False)]
+        }
+    )
+    reviewer_ids = fields.Many2many(
+        'res.users', string='Reviewers',
+        copy=False,
+        readonly=True,
+        states={
+            'draft': [('readonly', False)],
+            'confirmed': [('readonly', False)]
+        }
+    )
+    approval_date = fields.Datetime(
+        string='Approval Date',
+        copy=False,
+        readonly=True
+    )
+    approved_id = fields.Many2one(
+        'res.users',
+        string='Approved by',
+        copy=False,
+        readonly=True
+    )
+    company_id = fields.Many2one(
+        'res.company', string='Company',
+        required=True, readonly=True, states={'draft': [('readonly', False)]},
+        default=_get_default_company,
+    )
+    to_be_reviewed = fields.Boolean(
+        string='To be Reviewed'
+    )
+    kanban_state = fields.Selection(
+        [
+            ('normal', 'In Progress'),
+            ('on_hold', 'On Hold'),
+            ('done', 'Ready for next stage')
+        ],
+        'Kanban State',
+        track_visibility='onchange',
+        default='normal'
+    )
+    origin = fields.Char(
+        string='Source',
+        readonly=True,
+        states={
+            'draft': [('readonly', False)],
+            'confirmed': [('readonly', True)]
+        }
+    )
+    portal_published = fields.Boolean('In Portal', default=False)
+    user_id = fields.Many2one('res.users', string='Owner',
+                              default=lambda self: self.env.user,
+                              required=True)
+    date = fields.Date(
+        'Date',
+        default=lambda self: self._context.get(
+            'date', fields.Date.context_today(self)),
+        required=True
+    )
+
+    @api.model
+    def create(self, vals):
+        if vals.get('name', '/') == '/':
+            vals['name'] = self.env['ir.sequence'].next_by_code(
+                'business.requirement')
+        return super().create(vals)
+
+    @api.multi
+    def write(self, vals):
+        if vals.get('state'):
+            br_xml_id = self.env.ref(
+                'business_requirement.group_business_requirement_manager')
+            user = self.env.user
+            grps = self.env.user.groups_id.ids
```

Comment:
```suggestion
            grps = set(self.env.user.groups_id.ids)
```

Reply:
No need. There are no duplicates in that field. ORM ensures that.

Reply:
First comment updated

Reply:
Well, I would use `user.has_group` method instead, but this set optimization is not very decisive for the performance